### PR TITLE
Fix "Unable to access IdP links" 

### LINKED ISSUE #700: Unable to access Links (_links) in version 7 for an IdentityProvider when using GetIdentityProviderAsync/CreateIdentityProviderAsync
### Describe the bug?

I am trying to retrieve the metadata and ACS information of an IdentityProvider. From the documentation, it is supposed to appear in `_links` property and I also do see it in postman response in the following format: 

`"_links": {
        "metadata": {
            "href": "<url>/metadata.xml",
            "type": "application/xml",
            "hints": {
                "allow": [
                    "GET"
                ]
            }
        },
        "acs": {
            "href": "<url>",
            "type": "application/xml",
            "hints": {
                "allow": [
                    "POST"
                ]
            }
        },
        "users": {
            "href": "<url>/users",
            "hints": {
                "allow": [
                    "GET"
                ]
            }
        },
        "deactivate": {
            "href": "<url>/lifecycle/deactivate",
            "hints": {
                "allow": [
                    "POST"
                ]
            }
        }`

The readme(https://github.com/okta/okta-sdk-dotnet/tree/legacy-5.x-series#get-data-from-_links) talks about version 5 using the `GetProperty` Resource, but this class has been removed in version 7 and the Links property is returned as Null in the SDK response. Can you please provide a workaround as this is a critical functionality that I am trying to access from the API and I don't see any way.

### What is expected to happen?

A way to access Links property for an IdentityProvider object.

### What is the actual behavior?

Links is null in the SDK response 

### Reproduction Steps?

1. Use .Net SDK version 7. 
2. Call IIdentityProviderApi.GetIdentityProviderAsync(idpId), where idpId is the ID of the Identity provider
3. Check the IdentityProvider object returned in the response.

### Additional Information?

_No response_

### .NET Version

8.0.201

### SDK Version

7.0.5

### OS version

Darwin MacBook-Pro 22.6.0 Darwin Kernel Version 22.6.0: Tue Nov  7 21:40:08 PST 2023; root:xnu-8796.141.3.702.9~2/RELEASE_ARM64_T6000 arm64

### PR DESCRIPTION
**Pull request by laura-rodriguez**
- Update OAS with missing links
- Enable AdditionalProperties so links that are not defined in the spec are still accessible
- Add test to verify links are accessible
- Fix #700 

Files to review:
openapi3/management.yaml
Okta.Sdk/Model/IdentityProvider and IdentityProviderLinks
[Okta.Sdk.UnitTest/Api/IdentityProviderApiTests.cs](https://github.com/okta/okta-sdk-dotnet/pull/708/files#diff-25fa46f88fb0520704209c403af1c493fe8ba873296f760c05b71caaa6e9c0d9)

### CODE REVIEW

File: src/Okta.Sdk/Model/IdentityProvider.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ public bool ShouldSerializeLastUpdated()
         /// Gets or Sets Links
         /// </summary>
         [DataMember(Name = "_links", EmitDefaultValue = true)]
-        public LinksSelf Links { get; set; }
+        public IdentityProviderLinks Links { get; set; }
```
Comment (bryanapellanes-okta) [inline src/Okta.Sdk/Model/IdentityProvider.cs:128]: This is a breaking change right?  Will you update the version for this PR or are do you intend to submit another after merging this?

**Review by bryanapellanes-okta — COMMENTED** [review]
Quick question before approving.

File: src/Okta.Sdk/Model/IdentityProvider.cs (hunk 1 above)
Comment (laura-rodriguez) [reply src/Okta.Sdk/Model/IdentityProvider.cs:128]: Correct, I'm planning to address the release in a separate PR.

**Review by bryanapellanes-okta — APPROVED** [review]
👍

**Pull request merged**